linkup-cli/src/commands/uninstall.rs

```diff
@@ -1,43 +1,49 @@
 use std::{fs, process};
 
-use crate::{commands, linkup_dir_path, CliError};
+use crate::{commands, linkup_dir_path, linkup_exe_path, CliError, InstallationMethod};
 
 #[derive(clap::Args)]
 pub struct Args {}
 
 pub fn uninstall(_args: &Args) -> Result<(), CliError> {
     commands::stop(&commands::StopArgs {}, true)?;
 
+    let exe_path = linkup_exe_path();
+
+    log::debug!("Linkup exe path: {:?}", &exe_path);
+    match InstallationMethod::current() {
+        InstallationMethod::Brew => {
+            log::debug!("Uninstalling linkup from Homebrew");
+
+            process::Command::new("brew")
+                .args(["uninstall", "linkup"])
+                .stdin(process::Stdio::null())
+                .stdout(process::Stdio::null())
+                .stderr(process::Stdio::null())
+                .status()?;
+        }
+        InstallationMethod::Cargo => {
+            log::debug!("Uninstalling linkup from Cargo");
+
+            process::Command::new("cargo")
+                .args(["uninstall", "linkup-cli"])
+                .stdin(process::Stdio::null())
+                .stdout(process::Stdio::null())
+                .stderr(process::Stdio::null())
+                .status()?;
+        }
+        InstallationMethod::Manual => {
+            log::debug!("Uninstalling linkup");
+
+            fs::remove_file(&exe_path)?;
```

Comment:
Do you think we "should" remove any caddy builds there are locally in this case?

Reply:
If I get what you mean, it will happen since it is installed inside the `./linkup/bin` folder, and we delete the whole `.linkup` we delete on line 44.
One bigger problem that we have atm is related to brew upgrade if there is a new custom Caddy build, since we don't handle updates of Caddy on Homebrew.

I already [created a PR to remove the official one from being installed](https://github.com/mentimeter/homebrew-mentimeter/pull/3), but the upgrade won't do anything about it and the user will need to do a uninstall/install of local-dns to get the latest one.
[I'm looking to find a better way to handle this.](https://github.com/mentimeter/linkup/tree/augustoccesar/new-release-workflow)

Reply:
Smart, sounds great!